python/tvm/autotvm/tuner/xgboost_cost_model.py

```diff
@@ -321,10 +316,11 @@ def _get_feature(self, indexes):
 
         indexes = np.array(indexes)
         need_extract = [x for x in indexes if x not in fea_cache]
+        args = [(self.space.get(x), self.target, self.task) for x in need_extract]
```

Comment:
Doing this still needs serializing a lot of things. Did you test the performance before vs. after?